File: frontend-integration/flutter/models/terrain.dart
/// Matches backend Terrain entity.
class Terrain {
  final int id;
  final String name;
  final String type; // STADE | SPORT | SITE
  final String? city;
  final String? address;
  final double? latitude;
  final double? longitude;
  final int? capacity;
  final String? imageUrl;
  final String? description;

  Terrain({
    required this.id,
    required this.name,
    required this.type,
    this.city,
    this.address,
    this.latitude,
    this.longitude,
    this.capacity,
    this.imageUrl,
    this.description,
  });

  factory Terrain.fromJson(Map<String, dynamic> j) => Terrain(
        id: j['id'],
        name: j['name'],
        type: j['type'],
        city: j['city'],
        address: j['address'],
        latitude: (j['latitude'] as num?)?.toDouble(),
        longitude: (j['longitude'] as num?)?.toDouble(),
        capacity: j['capacity'],
        imageUrl: j['imageUrl'],
        description: j['description'],
      );
}

File: backend-tourisme-app/src/main/java/com/tourisme/tourisme_app/model/Terrain.java
package com.tourisme.tourisme_app.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "terrains")
@Data
public class Terrain {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TerrainType type;

    private String city;
    private String address;

    private Double latitude;
    private Double longitude;

    private Integer capacity;        // useful for stadiums
    private String imageUrl;

    @Column(length = 1000)
    private String description;
}
